synapse_data_warehouse/database_roles/V2.47.1__synapse_aggregate_database_roles.sql

```diff
@@ -0,0 +1,43 @@
+USE DATABASE {{ database_name }}; --noqa: JJ01,PRS,TMP
+
+-- Create "admin" database role which will own the synapse_aggregate objects
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ADMIN;
+
+-- Grant ownership of this database role to the proxy admin
+GRANT OWNERSHIP
+    ON DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ADMIN
+    TO ROLE {{ database_name }}_PROXY_ADMIN; --noqa: JJ01,PRS,TMP
+
+-- Grant usage of database role to the proxy admin
+GRANT DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ADMIN
+    TO ROLE {{ database_name }}_PROXY_ADMIN; --noqa: JJ01,PRS,TMP
+
+-- Create "analyst" role which will inherit the object-type-specific roles 
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ANALYST;
+
+-- Grant ownership of the analyst role to the admin role
+GRANT OWNERSHIP
+    ON DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ANALYST
+    TO DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ADMIN;
+
+-- Grant analyst role to the analyst account role
+GRANT DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ANALYST
+    TO ROLE {{ database_name }}_ANALYST; --noqa: JJ01,PRS,TMP
+
+-- Create a database role for table read privileges 
+CREATE OR REPLACE DATABASE ROLE SYNAPSE_AGGREGATE_TABLE_READ;
+-- Grant ownership to the admin database role
+GRANT OWNERSHIP
+	ON DATABASE ROLE SYNAPSE_AGGREGATE_TABLE_READ
+	TO DATABASE ROLE SYNAPSE_AGGREGATE_ALL_ADMIN;
+-- Grant usage to the analyst database role
```

Comment:
Two things from comparing with what's in prod:

<img width="877" alt="image" src="https://github.com/user-attachments/assets/505a8d43-e484-4efc-9807-a2ae78c351c6" />

1. USAGE of the analyst role should also be granted to the developer role
2. In the `synapse` schema, this role is granted to the analyst _account_ role, not the all_analyst _database_ role. I prefer the way you have it here (i.e. the usage privilege flows up to the account role through the database role), but if we move forward that way, we'll have to change it in the other schemas.

Reply:
> USAGE of the analyst role should also be granted to the developer role

Nice catch! You're right, this is pretty confusing to do correctly without a guide or list of to-dos to make sure we don't forget anything.

Could you clarify your point #2? I checked both the dev and prod analyst account roles and only saw inheritance of *ALL_ANALYST database roles.

Reply:
Please ignore point 2, I got mixed up with the roles!

Reply:
I made a new ticket for the bootstrap-a-schema template: https://sagebionetworks.jira.com/browse/SNOW-276